addons/mail/static/src/js/utils.js

```diff
@@ -45,21 +42,21 @@ function linkify(text, attrs) {
 
 function addLink(node, transformChildren) {
     if (node.nodeType === 3) {  // text node
-        return linkify(node.data);
+        return linkify(_.escape(node.data));
     }
     if (node.tagName === "A") return node.outerHTML;
     node.innerHTML = transformChildren();
     return node.outerHTML;
 }
 
 function stripHTML(node, transformChildren) {
-    if (node.nodeType === 3) return node.data;  // text node
+    if (node.nodeType === 3) return _.escape(node.data);  // text node
     if (node.tagName === "BR") return "\n";
     return transformChildren();
 }
 
 function inline(node, transform_children) {
-    if (node.nodeType === 3) return node.data;
+    if (node.nodeType === 3) return _.escape(node.data);
```

Comment:
Same question.

Reply:
because then:

`mailUtils.parseAndTransform('<div>&amp; &amp;amp;</div>', mailUtils.addLink)`

returns:

`<div>& &amp;</div>`

which is shown in browser as:

`& &`

and not original `& &amp;`

Reply:
I think ok for the change in `inline` then, it will convert `&` into `&amp;` which wasn't the case before, but if the output is considered as HTML and not as text, then it shouldn't matter because both `&amp;` and `&` would display the same.